src/actions/BlockActions.js

```diff
@@ -25,15 +25,78 @@ import TransactionActions from './TransactionActions';
 
 import GlobalReducer from '../reducers/GlobalReducer';
 
+/**
+ *
+ * @param {Object} targetBlock
+ * @param {Object|undefined} targetBlock
+ * @param {Object|undefined} targetBlockReward
+ */
+const getRewardDistribution = async (targetBlock, nextBlock) => {
+	if (!nextBlock) {
+		return null;
+	}
+
+	const { account: producerId, delegate: delegateId } = targetBlock;
+
+	const producer = {
+		type: 'Producer',
+		producer: undefined,
+		delegate: undefined,
+		producedByCommittee: false,
+	};
+
+	if (producerId) {
+		const account = await echo.api.getObject(producerId);
+		producer.producer = account ? account.name : undefined;
+	}
+
+	if (delegateId) {
+		const account = await echo.api.getObject(delegateId);
+		producer.delegate = account ? account.name : undefined;
+	}
+
+	const verifiers = nextBlock.prev_signatures
+		.map(async (s) => {
+			const res = {
+				type: 'Verifier',
+				producer: undefined,
+				delegate: undefined,
+				producedByCommittee: !!s._fallback,
+			};
+
+			if (s._producer) {
+				const id = `${ACCOUNT_OBJECT_PREFIX}.${s._producer}`;
+				const account = await echo.api.getObject(id);
+
+				res.producer = account ? account.name : undefined;
+			}
+
+			if (s._delegate) {
```

Comment:
why verifiers skip null-account, but producer set this to delegate account

Reply:
0 account means producer make message by himself, i removed delegate field in same case from table at producer type field